### ISSUE TITLE
query result error in norec mode

### ISSUE THREAD

**Issue by wanpengfei-git** [StarRocks/starrocks#5030]
### Steps to reproduce the behavior (Required)
```
DROP DATABASE IF EXISTS n0;
CREATE DATABASE n0;
USE n0;
admin set frontend config ("enable_decimal_v3" = "true");
CREATE TABLE t0 (c_0_0 BOOLEAN NOT NULL ,c_0_1 DECIMAL(3, 0)  NULL ) DUPLICATE KEY (c_0_0) DISTRIBUTED BY HASH (c_0_0) BUCKETS 3 properties("replication_num"="3") ;
CREATE TABLE t1 (c_1_0 DATETIME NOT NULL ) AGGREGATE KEY (c_1_0) PARTITION BY RANGE(c_1_0) (START ("2010-01-01") END ("2021-12-31") EVERY (INTERVAL 292 day)) DISTRIBUTED BY HASH (c_1_0) BUCKETS 3 properties("replication_num"="3") ;
INSERT INTO t0 VALUES (false, 375), (false, 725);
INSERT INTO t0 VALUES (false, 234), (false, 195);
INSERT INTO t1 VALUES ("2010-01-16 08:00:14"), ("2010-01-10 08:00:14");
INSERT INTO t1 VALUES ("2010-01-03 08:00:14"), ("2010-01-31 08:00:14");
CREATE MATERIALIZED VIEW v0 AS SELECT t1.c_1_0 FROM t1;
CREATE MATERIALIZED VIEW v0 AS SELECT t1.c_1_0 FROM t1;
INSERT INTO t0 VALUES (true, 712), (false, 452);
INSERT INTO t1 VALUES ("2010-01-05 08:00:14");
INSERT INTO t1 VALUES ("2010-01-06 08:00:14");
INSERT INTO t0 VALUES (false, NULL), (true, 656);
INSERT INTO t1 VALUES ("2010-01-28 08:00:14"), ("2010-01-03 08:00:14");
INSERT INTO t1 VALUES ("2010-01-10 08:00:14"), ("2010-01-31 08:00:14");
INSERT INTO t0 VALUES (true, 786);
INSERT INTO t1 VALUES ("2010-01-10 08:00:14");
INSERT INTO t0 VALUES (true, 679), (false, 758), (false, 676);

mysql> WITH with_t_0 as (SELECT t1.c_1_0 FROM t1 WHERE EXISTS (SELECT t1.c_1_0 FROM t1 WHERE (NOT (false))) ) SELECT COUNT(*) FROM t1, with_t_0 WHERE (NOT ((NOT (true))));
+----------+
| count(*) |
+----------+
|      147 |
+----------+
1 row in set (0.02 sec)

mysql> WITH with_t_0 as (SELECT t1.c_1_0 FROM t1 WHERE EXISTS (SELECT t1.c_1_0 FROM t1 WHERE (NOT (false))) ) SELECT SUM(count) FROM (SELECT CAST((CAST((NOT ((NOT (true)))) AS BOOLEAN) = true) AND (CAST((NOT ((NOT (true)))) AS BOOLEAN) IS NOT NULL) AS INT) as count FROM t1, with_t_0 ) t;
+------------+
| sum(count) |
+------------+
|         49 |
+------------+

```

### StarRocks version (Required)
 - You can get the StarRocks version by executing SQL `select current_version()`
 - branch-2.2 ASAN
 - mysql> select current_version();
+------------------------------+
| current_version()            |
+------------------------------+
| SQLANCER_01_ASAN 145a79c |
+------------------------------+

**Comment by ZiheLiu** [comment]
Fixed by #5022.

**Comment by lvchenyang-maker** [comment]
```
mysql> WITH with_t_0 as (SELECT t1.c_1_0 FROM t1 WHERE EXISTS (SELECT t1.c_1_0 FROM t1 WHERE (NOT (false))) ) SELECT COUNT(*) FROM t1, with_t_0 WHERE (NOT ((NOT (true))));
+----------+
| count(*) |
+----------+
|      147 |
+----------+
1 row in set (0.01 sec)

mysql> WITH with_t_0 as (SELECT t1.c_1_0 FROM t1 WHERE EXISTS (SELECT t1.c_1_0 FROM t1 WHERE (NOT (false))) ) SELECT SUM(count) FROM (SELECT CAST((CAST((NOT ((NOT (true)))) AS BOOLEAN) = true) AND (CAST((NOT ((NOT (true)))) AS BOOLEAN) IS NOT NULL) AS INT) as count FROM t1, with_t_0 ) t;
+------------+
| sum(count) |
+------------+
|         49 |
+------------+
1 row in set (0.02 sec)
```
version
```
+------------------------------------------+
| current_version()                        |
+------------------------------------------+
| PIPELINE_SQLANCER_MASTER_RELEASE b7a7e39 |
+------------------------------------------+
```

**Comment by ZiheLiu** [comment]
@lvchenyang-maker This has already been fixed by #5022, but has not yet been cherry-pick to branch-2.2.
